### PR TITLE
Allow for whitelisted flags to be passed to the Dart VM

### PR DESCRIPTION
**Pull request by bkonyi**
Fixes part of https://github.com/flutter/flutter/issues/32176

### CODE REVIEW

**Comment by bkonyi** [comment]
cc @devoncarew @jacob314 @kenzieschmoll @terrylucas

File: shell/common/switches.cc [hunk 1]
```diff
@@ -36,6 +37,22 @@ struct SwitchDesc {
 #define DEF_SWITCHES_END };
 // clang-format on
 
+#if FLUTTER_RUNTIME_MODE != FLUTTER_RUNTIME_MODE_RELEASE && \
+    FLUTTER_RUNTIME_MODE != FLUTTER_RUNTIME_MODE_DYNAMIC_RELEASE
+
+// List of common and safe VM flags to allow to be passed directly to the VM.
+static const std::string gDartFlagsWhitelist[] = {
```
Comment (bkonyi) [inline shell/common/switches.cc]: The current whitelist is just composed of some flags that I thought would be useful and safe. I'm happy to add/remove from this list based on feedback.

File: shell/platform/android/io/flutter/app/FlutterActivityDelegate.java [hunk 2]
```diff
@@ -312,6 +312,12 @@ public void onConfigurationChanged(Configuration newConfig) {}
         if (intent.getBooleanExtra("verbose-logging", false)) {
             args.add("--verbose-logging");
         }
+        // NOTE: all flags provided with this argument are subject to filtering
+        // based on a whitelist in shell/common/switches.cc. If any flag provided
+        // is not present in the whitelist, the engine will immediately shut down.
```
Comment (chinmaygarde) [inline shell/platform/android/io/flutter/app/FlutterActivityDelegate.java]: "the process will be immediately terminated."

**Comment by bkonyi** [comment]
> Can you add a unit test for this in the shell_unittests target?

Testing that the whitelist check catches bad flags will cause the process to exit. Did we just want to check that whitelisted flags aren't dropped?

**Comment by chinmaygarde** [comment]
You can use death tests in the Google test harness https://github.com/google/googletest/blob/master/googletest/docs/advanced.md#death-tests

File: shell/platform/android/io/flutter/app/FlutterActivityDelegate.java (hunk 2 above)
Comment (bkonyi) [reply shell/platform/android/io/flutter/app/FlutterActivityDelegate.java]: Done.

**Comment by bkonyi** [comment]
> You can use death tests in the Google test harness https://github.com/google/googletest/blob/master/googletest/docs/advanced.md#death-tests

Oh cool, I didn't know that existed. Thanks for guiding me in the right direction :-) I've added a couple of tests.

**Pull request merged**